mono/mini/method-to-ir.c

```diff
@@ -3759,6 +3759,15 @@ mono_emit_load_got_addr (MonoCompile *cfg)
 	MONO_ADD_INS (cfg->bb_exit, dummy_use);
 }
 
+static gboolean
+method_does_not_return (MonoMethod *method)
```

Comment:
Could we use better logic than method name? I guess checking for no `ret` should be sufficient for what we need? 

Note: [DoesNotReturn] is C# compiler flow-analysis attribute and we cannot rely on it tools like linker will try to remove it.

Reply:
Checking for no ret would involve parsing the IL for every method we want to inline, so if DoesNotReturn is not usable, then this hack would probably need to stay.

Reply:
Maybe it should be implemented in Roslyn (detect and mark such methods with DoesNotReturn automatically?)

Reply:
That was discussed and rejected by Roslyn folks

Reply:
This is just an internal class, if something changes, it will be pretty easy to detect the breakage. Can also check that it throws when we compile the called method.